orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CancelStageHandler.kt

```diff
@@ -37,6 +40,21 @@ class CancelStageHandler(
 
   override fun handle(message: CancelStage) {
     message.withStage { stage ->
+      if (stage.status == RUNNING) {
```

Comment:
Lets add a javadoc comment as to why this is necessary (and particularly why it's safe to re-run -- something something we check for cancel immediately)